action.yml

```diff
@@ -4,15 +4,44 @@ description: 'Install Cosign and put it on your path'
 # This is pinned to the last major release, we have to bump it for each action version.
 # TODO: We should version the action separately from the binary, and add a paramter for which cosign
 # version to install.
+inputs:
+  cosign-release:
+    description: 'Cosign release version to use in the actions.'
+    required: false
+    default: 'v0.1.0'
 runs:
   using: "composite"
   steps:
     - run: curl -LO https://storage.googleapis.com/cosign-releases/v0.1.0/cosign
       shell: bash
     # We verify the version against a SHA **in the published action itself**, not in the GCS bucket.
     - run: |
-        sha=$(shasum -a 256 cosign | cut -d' ' -f1);
-        if [[ $(shasum -a 256 cosign | cut -d' ' -f1) != "fd2e09bb1ca5f5342227b85a4d3b1c498df1ab2439cafb3636381bb82ad29cea" ]]; then exit 1; fi
+        shaBootstrap=$(shasum -a 256 cosign | cut -d' ' -f1);
+        if [[ $shaBootstrap != "fd2e09bb1ca5f5342227b85a4d3b1c498df1ab2439cafb3636381bb82ad29cea" ]]; then exit 1; fi
+
+        semver='^v([0-9]+\.){0,2}(\*|[0-9]+)$'
+        if [[ ${{ inputs.cosign-release }} =~ $semver ]]; then
+          echo "INFO: Custom Cosign Version ${{ inputs.cosign-release }}"
+        else
+          echo "ERROR: Unable to validate cosign version: '${{ inputs.cosign-release }}'"
+          exit 1
+        fi
+
+        # Download custom cosign
+        curl -L https://storage.googleapis.com/cosign-releases/${{ inputs.cosign-release }}/cosign -o cosign_${{ inputs.cosign-release }}
+        shaCustom=$(shasum -a 256 cosign_${{ inputs.cosign-release }} | cut -d' ' -f1);
+
+        # check if is the same hash means it is the same release
+        if [[ $shaCustom != $shaBootstrap ]];
+        then
+          chmod +x cosign
+          curl -LO https://github.com/sigstore/cosign/releases/download/${{ inputs.cosign-release }}/cosign.sig
+          curl -LO https://raw.githubusercontent.com/sigstore/cosign/main/.github/workflows/cosign.pub
```

Comment:
One nit here - I think this URL might have to change depending on the release as well, in case we rotate the key? 

Basically:

`https://raw.githubusercontent.com/sigstore/cosign/${{ inputs.cosign-release }}/.github/workflows/cosign.pub` instead of the `main`.

Reply:
makes sense, thanks for your review and feedback

Reply:
Perfec!